tests/virt_autotest/reboot_and_wait_up.pm

```diff
@@ -81,7 +81,9 @@ sub reboot_and_wait_up {
         reset_consoles;
 
         #wait boot finish and relogin
+        set_var('GUI_INSTALL', '1');
```

Comment:
Do we need this ```GUI_INSTALL``` ?

Reply:
Agree, the purpose of this var is clear, but name is a little bit misleading for some situations, eg host is installed by AUTOYAST, in the afterwards reboot_and_wait_upxx, it will be set to GUI_INSTALL, which is opposite in logic.

Reply:
The logic:  there's a "GUI installation mode value" which is 210.    In "afterwards reboot_and_wait_up" function  we always use this value no matter if the actual installation is GUI or autiyast.  There is no first login or second ...

What would you guys suggest to change?

Reply:
I do no think there is need to differentiate first login or second or third, for example, reboot_and_wait_up_normal can be called anywhere, and for host upgrade scenario, detecting grub after upgrade is meaningful (before host_upgrade_step3_run). 
Did you invent it or is being used widely in other tests already ?

Reply:
It's a workaround. See https://progress.opensuse.org/issues/128201#note-35 and related comments in  that tickets.

Reply:
I do not quite understand your change here. There is no need to change this part at all. And it seems that if you set GUI_INSTALL=0, then there will be actually no grub menu checking at all (because 1 second is not enough to capture anything and the system might not load grub menu at that time). I do not think this is acceptable. We use autoyast to install host which does not mean there is no need to detecting grub menu after install.

Reply:
> 
> What would you guys suggest to change?

I suggest use `caller`.

Reply:
> > What would you guys suggest to change?
> 
> I suggest use `caller`.

Maybe just add another extra line in login_console.pm, so do not modify any existing lines, for example, add following after two reset_consoles:
```
if (caller xxx and AUTOYAST and other required constraints) {
    reboot immediately
}
```
???
Straightforward and easy.

Reply:
@tonyyuan1 If you finally decide to use current way -- using a var to differentiate, please update the variable name . See more details in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17529/files#r1286533531.

Reply:
Do you mean "GUI_INSTALL"?
Note: it's just a temporary var.  What name do you recommend?

Reply:
Yes. Any name not resulting in conflicting info is okay, since this is a workaround anyway for calling login_console,  after autoyast installation during which the grub menu already passed, but any other cases for calling login_console should keep the grub check(eg reboot_and_waitup_xx and original graphical host installation).     The name can be, for example,  KEEP_INITIAL_GRUB_CHECK.

Please note that if  only var name changed from `GUI_INSTALL`, and any other code unchanged in the PR, you will need to set the var to 1 in all testsuites that keep original graphical installation. Otherwise the first login_console after host installation, will mostly has no grub screen matching. So, IMHO, a minor improvement is to SET this var only during AUTOYAST installation, unset it afterwards. Then anywhere else won't need setting for this var at all. Up to you to decide. @Julie-CAO  FYI. Please pay attention to this point in the final OSD deployment during 15sp6.

Reply:
Hi @alice-suse, Based on current logic, we don't need add `GUI_INSTALL` setting to any test suite. It is set in `reboot_and_wait_up.pm`. Only tests with autoyast installation will skip grub check in login_console.pm.

Reply:
> Hi @alice-suse, Based on current logic, we don't need add `GUI_INSTALL` setting to any test suite. It is set in `reboot_and_wait_up.pm`. Tests with autoyast installation will skip grub check in login_console.pm.

1. This var ```GUI_INSTALL``` is really confusing. If author wants use variable to differentiate first boot after autoyast from other boots, a better name conveying correct message is required. For example, "reboot_after_autoyast" or similar. I do think the name of this var is very acceptable.
2. If autoyast install, there will be only grub checking once in any case, namely the inner check_screen in unless block.

Reply:
> > Hi @alice-suse, Based on current logic, we don't need add `GUI_INSTALL` setting to any test suite. It is set in `reboot_and_wait_up.pm`. Tests with autoyast installation will skip grub check in login_console.pm.
> 

@Julie-CAO  Hi Julie, I hope you get my precondition for this comment -- "you will need to set the var to 1 in all testsuites **that keep original graphical installation**".


> 1. This var `GUI_INSTALL` is really confusing. If author wants use variable to differentiate first boot after autoyast from other boots, a better name conveying correct message is required, just like `UPGRADE_AFTER_REBOOT` and `REBOOT_AFTER_UPGRADE` for offline host upgrade in login_console.pm. For example, "reboot_after_autoyast" or similar. I do think the name of this var is very acceptable if author only wants to differentiate first boot after autoyast from other boots. And I do not think we should bother GUI install (literally) at all.

Agree. In https://progress.opensuse.org/issues/128201#note-35, I initially suggested a similar name ,  `FIRST_REBOOT_AFTER_AUTOYAST_INST` .  I also sugested that this var was better to be set in autoyast installation module, and unset it after the first boot grub check in login_console.

Reply:
@tonyyuan1 Just summarize for you. New name suggestions for `GUI_INSTALL` -- `REBOOT_AFTER_AUTOYAST`, `FIRST_REBOOT_AFTER_AUTOYAST_INST`, or `KEEP_INITIAL_GRUB_CHECK` . Seems good to pick one from the first two options.

Reply:
After reading all of comments here, It seems that:
A VAR name just can't confuse people. It might be that  the logic behind it is a bit difficult to understand.
'GUI_INSTALL' is short for "GUI installation mode" which is related to the logic.  Changing it makes it irrelevant to logic. Therefore, I add a inline comment to explain them. Let me know if it look better to you.

Reply:
> After reading all of comments here, It seems that: A VAR name just can't confuse people. It might be that the logic behind it is a bit difficult to understand. 'GUI_INSTALL' is short for "GUI installation mode" which is related to the logic. Changing it makes it irrelevant to logic. Therefore, I add a inline comment to explain them. Let me know if it look better to you.

It depends on why you needs this variable. If it just differentiates first boot after autoyast from other boots, it has to be renamed.

Reply:
@tonyyuan1 Does your ```GUI_INSTALL``` refer to previous step-by-step text installation ? If this is the case, I do not think it is very good. It can kind of mix up text mode and real graphical interface, and I think you should consider other variable names. And using ```GUI_INSTALL``` does not reflect what you are trying to solve, and it seems that using it indicates a situation of different nature.

Reply:
To me, it seems better with the comment explanation. 
Given that Wayne still has questions and suggestions on this, in addition to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17529/files#r1295552604,  I will hold on the next move, until you guys make things clear to each other and get agreement. I support NECESSARY items with minor effort to be done in this PR before merge, while   others maybe not.